src/main/java/uk/gov/companieshouse/bankruptofficersearch/api/request/OracleQueryDaoImpl.java

```diff
@@ -69,7 +69,7 @@ public ScottishBankruptOfficerDetailsEntity getScottishBankruptOfficer(final Str
                 return null;
             }
 
-            throw new OracleQueryApiException(ex.getStatusCode(), ex.getStatusText());
+            throw new OracleQueryApiException(ex.getMessage(), ex.getCause());
```

Comment:
Can logging be added here so it's clear in the logs what's happening?
```
try {
    return restTemplate.getForObject(oracleQueryApiUrl + uri, ScottishBankruptOfficerDetailsEntity.class);
} catch (HttpClientErrorException e) {
    map.put("status_code", e.getStatusCode());

    if (e.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
        LOGGER.debug("Oracle query API returned not found", map); 
        return null;
    }

    LOGGER.error(e, map);
    throw new OracleQueryApiException(e.getMessage(), e);
}
```

Does HttpClientErrorException always have a status code? What if an invalid formatted/missing URL was passed in? This will throw a NullPointerException if status code can be null?

Reply:
Added extra logging
Tried to add a null check for the statusCode but Sonar complains it always evaluates to true